src/TSTransformer/nodes/statements/transformForOfStatement.ts

```diff
@@ -435,6 +437,50 @@ function getLoopBuilder(state: TransformState, node: ts.Node, type: ts.Type): Lo
 	}
 }
 
+export function transformForOfRangeMacro(state: TransformState, node: ts.ForOfStatement): luau.List<luau.Statement> {
+	// we proved this before
+	assert(ts.isCallExpression(node.expression));
+
+	if (!ts.isVariableDeclarationList(node.initializer)) {
+		DiagnosticService.addDiagnostic(errors.noRangeMacroExpressionInitializer(node.initializer));
+		return luau.list.make();
+	}
```

Comment:
Why isn't this allowed? It could be run before the loop, and then just use `luau.tempId()` to discard the loop counter.

Reply:
(Got my stuff a bit wrong, doing an identifier here means setting the loop stuff to that variable before each iteration. That's implemented now by moving to `transformForInitializer`)